components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/TransactionHandlerComponent.java

```diff
@@ -0,0 +1,172 @@
+            if (null != scheduledPeriodObject) {
+                scheduledPeriodStr = scheduledPeriodObject.toString();
+            }
+        }
+        try {
+            return Integer.parseInt(scheduledPeriodStr);
+        } catch (NumberFormatException e) {
+            return DEFAULT_SCHEDULED_PERIOD_INTERVAL;
+        }
+    }
+
+    /**
+     * Schedule the transaction writer task to write the current transaction count to the database at a fixed rate.
+     *
+     * @param transactionStore - transactionStore instance.
+     */
+    private void scheduleTransactionWriterTask(TransactionStore transactionStore) {
+        ScheduledExecutorService trnCountWriterTaskScheduler = Executors.newSingleThreadScheduledExecutor();
+        int scheduledPeriod = getTransactionCountScheduledPeriod();
+
+        trnCountWriterTaskScheduler.scheduleAtFixedRate(() -> {
+            try {
+                transactionStore.addTransaction();
+            } catch (TransactionException e) {
+                LOG.warn("Could not persist the transaction count: ", e);
+            }
+        }, scheduledPeriod, scheduledPeriod, TimeUnit.SECONDS);
+    }
+
+    private String generateRandomId() {
+        return UUID.randomUUID().toString();
+    }
+
+    /**
+     * Check whether datasource is configured for transaction counting utility.
+     *
+     * @param dataSourceService dataSourceService instance.
+     * @return - true if datasource defined.
+     * @throws DataSourceException  - when DataSource is not an RDBMS data source.
+     * @throws TransactionException - when no DataSource is available for transaction counting utility.
+     */
+    public DataSource getTransactionDataSource(DataSourceService dataSourceService)
+            throws DataSourceException, TransactionException {
+
+        String dataSourceId = getTransactionDatasourceId();
+        CarbonDataSource dataSource = dataSourceService.getDataSource(dataSourceId);
+        if (dataSource == null) {
+            throw new TransactionException("DataSource " + dataSourceId + " is not available for transaction.");
+        }
+        Object transactionDataSourceObject = dataSource.getDSObject();
+        if (!(transactionDataSourceObject instanceof DataSource)) {
+            throw new DataSourceException("DataSource is not an RDBMS data source.");
+        }
+        return (DataSource) transactionDataSourceObject;
+    }
+
+    /**
+     * Get the dataSource configured for transaction counting utility.
+     *
+     * @return - datasource id.
+     * @throws TransactionException - when a dataSource is not configured for transaction counting utility.
+     */
+    private String getTransactionDatasourceId() throws TransactionException {
+        String dataSourceId = System.getProperty(TRANSACTION_DATA_SOURCE);
+        if (StringUtils.isEmpty(dataSourceId)) {
+            dataSourceId = System.getenv(TRANSACTION_DATA_SOURCE);
```

Comment:
Do we need this much of flexibility of reading this from system env .. Doesn't toml only suffice.

Reply:
I think we do not need to read from the env/sys variables since, it is highly unlikely that they might change across environments

Reply:
Changed the code to read the data source id only from deployment.toml.